main/cron/remind_course_expiration.php

```diff
@@ -0,0 +1,120 @@
+    " AS certificate ON category.id = certificate.cat_id
+    INNER JOIN ".Database::get_main_table(TABLE_MAIN_SESSION).
+    " AS session ON category.session_id = session.id
+    WHERE session.date_end BETWEEN '$today' AND '$expirationDate' AND category.session_id IS NOT NULL";
+
+$sessionId = 0;
+$userIds = array();
+$sessions = array();
+$result = Database::query($query);
+
+while ($row = Database::fetch_array($result)) {
+    if ($sessionId != $row['session_id']) {
+        $sessionId = $row['session_id'];
+        $userIds = array();
+    }
+    if (!is_null($row['user_id'])) {
+        array_push($userIds, $row['user_id']);
+    }
+    $sessions[$sessionId] = $userIds;
+}
+
+$usersToBeReminded = array();
+
+foreach ($sessions as $sessionId => $userIds) {
+    $userId = 0;
+    $userIds = $userIds ? " AND id_user NOT IN (".implode(",", $userIds).")" : null;
+    $query = "SELECT sessionUser.id_session, sessionUser.id_user, session.name, session.date_end FROM ".
+        Database::get_main_table(TABLE_MAIN_SESSION_USER)." AS sessionUser
+        INNER JOIN ".Database::get_main_table(TABLE_MAIN_SESSION).
+        " AS session ON sessionUser.id_session = session.id
+        WHERE id_session = $sessionId$userIds";
+    $result = Database::query($query);
+    while ($row = Database::fetch_array($result)) {
+        $usersToBeReminded[$row['id_user']][$row['id_session']] =
+            array(
+                'name' => $row['name'],
+                'date_end' => $row['date_end']
+            );
+    }
+}
+
+if ($usersToBeReminded) {
+    global $_configuration;
+    $today = date_create($today);
+    $platformLanguage = api_get_setting("platformLanguage");
+    $subject = sprintf(
+        get_lang("MailCronCourseExpirationReminderSubject", null, $platformLanguage),
+        api_get_setting("Institution")
+    );
+    $administrator = array(
+        'completeName' => api_get_person_name(
+            api_get_setting("administratorName"),
+            api_get_setting("administratorSurname"),
+            null,
+            PERSON_NAME_EMAIL_ADDRESS
+        ),
+        'email' => api_get_setting("emailAdministrator")
+    );
+    echo "\n======================================================================\n\n";
+    foreach ($usersToBeReminded as $userId => $sessions) {
+        $user = api_get_user_info($userId);
+        $userCompleteName = api_get_person_name(
+            $user['firstname'],
+            $user['lastname'],
+            null,
+            PERSON_NAME_EMAIL_ADDRESS
+        );
+        foreach ($sessions as $session) {
+            $daysRemaining = date_diff($today, date_create($session['date_end']));
+            $body = sprintf(
+                get_lang('MailCronCourseExpirationReminderBody', null, $platformLanguage),
+                $userCompleteName,
+                $session['name'],
+                $session['date_end'],
+                $daysRemaining->format("%d"),
+                $_configuration['root_web'],
```

Comment:
I'll have to hint you on how to support multi-url here ($_configuration['root_web'] doesn't).